package/MDAnalysis/lib/formats/xdrlib.pyx

```diff
@@ -0,0 +1,653 @@
+
+cdef enum:
+    EOK = 0
+    EHEADER = 1
+    ESTRING = 2
+    EDOUBLE = 3
+    EINTEGER = 4
+    EFLOAT = 5
+    EUNSIGNEDINTEGER = 6
+    ECOMPRESSION = 7
+    ECLOSE = 8
+    EMAGIC = 9
+    EMEMORY = 10
+    EENDOFFILE = 11
+    ENOTFOUND = 11
+
+error_message = ['OK', 'header', 'string', 'double', 'integer',
+                 'float', 'unsigned integer', 'compression',
+                 'close', 'magic', 'memory', 'endoffile',
+                 'notfound']
+
+import cython
+import numpy as np
+from os.path import exists
+from collections import namedtuple
+
+np.import_array()
+
+ctypedef np.float32_t DTYPE_T
+DTYPE = np.float32
+cdef int DIMS = 3
+cdef int HASX = 1
+cdef int HASV = 2
+cdef int HASF = 4
+
+
+cdef class _XDRFile:
+    """Base python wrapper for gromacs-xdr formats
+
+    This class can be similar to the normal file objects in python. The read()
+    function will return a frame and all information in it instead of a single
+    line. Additionally the context-manager protocoll is supported as well.
+
+    Parameters
+    ----------
+    fname: str
+        The filename to open.
+    mode: ('r', 'w')
+        The mode in which to open the file, either 'r' read or 'w' write
+
+    Raises
+    ------
+    ValueError
+        Wrong mode given
+    IOError
+        Couldn't read the file
+
+    Note
+    ----
+    This class can't be initialized use one of the subclasses XTCFile, TRRFile
+    """
+    cdef readonly int n_atoms
+    cdef int is_open
+    cdef int reached_eof
+    cdef XDRFILE *xfp
+    cdef readonly fname
+    cdef int current_frame
+    cdef str mode
+    cdef np.ndarray box
+    cdef np.ndarray _offsets
+    cdef int _has_offsets
+
+    def __cinit__(self, fname, mode='r'):
+        self.fname = fname
+        self.is_open = False
+        self.open(self.fname, mode)
+
+    def __dealloc__(self):
+        self.close()
+
+    def close(self):
+        """Close the open XDR file
+
+        Raises
+        ------
+        IOError
+            If the TRR file can't be closed for some reason
+        """
+        cdef int res = 1
+        if self.is_open:
+            res = xdrfile_close(self.xfp)
+            self.is_open = False
+            if res != 0:
+                raise IOError('Couldn\'t close file: {}, Error = XDRLIB-{}'.format(
+                    self.fname, error_message[res]))
+        return True
+
+    def open(self, fname, mode):
+        """Open a XDR file
+
+        If another XDR file is currently opened it will be closed
+
+        Parameters
+        ----------
+        fname: str
+            The filename to open.
+        mode: ('r', 'w')
+            The mode in which to open the file, either 'r' read or 'w' write
+
+        Raises
+        ------
+        ValueError
+            Wrong mode given
+        IOError
+            Couldn't read the file
+        """
+        if self.is_open:
+            self.close()
+        cdef int return_code = 1
```

Comment:
It is used in line 197 and 199. I can move the declaration of return_code down or even remove it completely since it can be a python object without problems.